Source/WebInspectorUI/UserInterface/Views/DOMTreeElement.js

```diff
@@ -2167,17 +2169,28 @@ WI.DOMTreeElement = class DOMTreeElement extends WI.TreeElement
         }
     }
 
-    _handleLayoutFlagsChanged(event)
+    _updateUnrenderedNodeDimming()
     {
-        this.listItemElement?.classList.toggle("rendered", this.representedObject.layoutFlags.includes(WI.DOMNode.LayoutFlag.Rendered));
+        let renderedAndDimmingEnabled = !WI.settings.domTreeDimsUnrenderedNodes.value
```

Comment:
I think rather than have every `WI.DOMTreeElement` listen for this change, maybe we have the `WI.DOMTreeOutline` (or even the `WI.DOMTreeContentView`) listen for it and set a top-level CSS property that is then used inside
```
.content-view.dom-tree.indicate-rendered .tree-outline.dom li:not(.rendered) > span > *:not(.html-comment)
                      ^^^^^^^^^^^^^^^^^^
```